datumaro/plugins/splitter.py

```diff
@@ -129,37 +152,74 @@ def _group_by_attr(items):
         Returns:
             by_attributes: dict of { combination-of-attrs : list of index }
         """
+
+        # float--> numerical, others(int, string, bool) --> categorical
+        def _is_number(value):
+            if isinstance(value, str):
+                try:
+                    float(value)
+                    return True
+                except ValueError:
+                    return False
+            elif isinstance(value, float):
```

Comment:
```suggestion
            elif isinstance(value, (int, float)):
```

Reply:
Check if you can get any use for `datumaro.util.cast`.

Reply:
'int' is regarded as categorical at the moment. 
That's why I checked 'float' only. 
For the `datumaro.util.cast`, I'll check it.